File: LeetCode/19. Remove Nth Node From End of List.cpp
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode *node1 = head, *node2 = head;
        while (n--) node1 = node1->next;
        if (!node1) return head->next;
        while (node1->next) node1 = node1->next, node2 = node2->next;
        ListNode *temp = node2->next;
        node2->next = temp->next;
        delete(temp);
        return head;
    }
};

File: neetcode-150/linkedlist/19. Remove Nth Node From End of List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode curr = head, prev = head;
        while (n-- > 0)
            curr = curr.next;
        if (curr == null) return head.next;
        while (curr.next != null)
        {
            prev = prev.next;
            curr = curr.next;
        }
        ListNode temp = prev.next;
        prev.next = temp.next;
        return head;
    }
}